### ISSUE TITLE
[bug] Clear-text direct messages are logged when log-level is either info or debug

### ISSUE THREAD

**Issue by nachtlied1** [superseriousbusiness/gotosocial#1412]
### Describe the bug with a clear and concise description of what the bug is.

Direct message content is logged in clear-text to `GoToSocial` application logs when the log level is set to `INFO` or `DEBUG`. This appears to happen for the following `func` / `level` combinations at minimum (there may be more, but these are the ones I've observed so far):

```
func=processing.(*processor).ProcessFromClientAPI level=INFO

func=federation.(*federatingActor).Send level=INFO

func=federatingdb.(*federatingDB).NewID level=DEBUG

func=federatingdb.(*federatingDB).Create level=DEBUG

func=processing.(*processor).ProcessFromFederator level=INFO
```

On an instance with more than one user, this effectively means that anyone with access to the application logs could read other users' direct messages in clear-text just by accessing and reading the logs.

### What's your GoToSocial Version?

0.6.0 git-f9e5ec9

### GoToSocial Arch

arm64 Docker

### Browser version

All

### What happened?

While reviewing my server's GoToSocial application logs, I noticed that my direct messages (both sent and received) were logged in clear-text at different log levels (`INFO` and `DEBUG`).

Example log entry for a received DM (clear-text DM content replaced by `[REDACTED DIRECT MESSAGE TEXT]`):
```
Feb  1 19:34:47 720d5c5b25c5 gotosocial[1]: timestamp="01/02/2023 19:34:47.849" func=federatingdb.(*federatingDB).Create level=DEBUG create="{\"@context\":\"https://www.w3.org/ns/activitystreams\",\"atomUri\":\"[REDACTED]\",\"attachment\":[],\"attributedTo\":\"[REDACTED]\",\"cc\":[],\"content\":\"\\u003cp\\u003e\\u003cspan class=\\\"h-card\\\"\\u003e\\u003ca href=\\\"[REDACTED]\\\" class=\\\"u-url mention\\\"[REDACTED DIRECT MESSAGE TEXT]",\"conversation\":\"tag:[REDACTED],2023-02-01:objectId=[REDACTED]:objectType=Conversation\",\"id\":\"[REDACTED]\",\"inReplyTo\":\"[REDACTED]\",\"inReplyToAtomUri\":\"[REDACTED]\",\"published\":\"2023-02-01T19:34:47Z\",\"replies\":{\"first\":{\"items\":[],\"next\":\"[REDACTED]\",\"partOf\":\"[REDACTED]\",\"type\":\"CollectionPage\"},\"id\":\"[REDACTED]\",\"type\":\"Collection\"},\"sensitive\":false,\"tag\":{\"href\":\"[REDACTED]\",\"name\":\"[REDACTED]\",\"type\":\"Mention\"},\"to\":\"[REDACTED]\",\"type\":\"Note\",\"url\": ...
```

Example log entry for a sent DM (clear-text DM content replaced by `[REDACTED DIRECT MESSAGE TEXT]`):

```
Feb  2 11:47:51 720d5c5b25c5 gotosocial[1]: timestamp="02/02/2023 11:47:51.476" func=processing.(*processor).ProcessFromClientAPI level=INFO activityType=Create objectType=Note fromAccount=[REDACTED] model="&{ID:01GR8Z8MS8HNV0EN8AD0DJZJWN CreatedAt:2023-02-02 11:47:50.184505919 +0000 UTC m=+167262.698495200 UpdatedAt:2023-02-02 11:47:50.184506001 +0000 UTC m=+167262.698495274 URI:[REDACTED] URL:[REDACTED] Content:<p><span class=\"h-card\"><a href=\"[REDACTED]\" class=\"u-url mention\" rel=\"nofollow noreferrer noopener\" target=\"_blank\">@<span>[REDACTED]</span></a></span> [REDACTED DIRECT MESSAGE TEXT]</p> AttachmentIDs:[] Attachments:[] TagIDs:[] Tags:[] MentionIDs:[...] Mentions:[...] EmojiIDs:[] Emojis:[] Local:[REDACTED] AccountID:[REDACTED] Account:<nil> AccountURI:[REDACTED] InReplyToID:[REDACTED] InReplyToURI:[REDACTED] InReplyToAccountID:[REDACTED] InReplyTo:<nil> InReplyToAccount:<nil> BoostOfID: BoostOfAccountID: BoostOf:<nil> BoostOfAccount:<nil> ContentWarning: Visibility:direct Sensitive:0x4000e149a9 Language:en CreatedWithApplicationID:[REDACTED] CreatedWithApplication:<nil> ActivityStreamsType:Note Text:[REDACTED DIRECT MESSAGE TEXT] Pinned:0x4000e153c7 ...
```

### What you expected to happen?

Logging clear-text direct messages is probably fine (although not ideal) if it's on a single-user instance.

However, this is likely a privacy concern on servers with more than one user.  As a mitigation, perhaps direct message content could be obfuscated in the application logs?

Direct message content obfuscation could be either a non-configurable or configurable option. I can see some pros/cons of both options, but the actual decision is probably best left up to you...

### How to reproduce it?

1. Start the `GoToSocial` service with `GTS_LOG_LEVEL='debug'`.  This will also capture anything with log level `INFO`.
2. Send a direct message to someone from an account on the server
3. Open the `GoToSocial` application log and look for the log entries that correspond with the direct message
    * (There should be at least 3-4 of these)
    * Look for log fields like `content`, `Content`, `Text`.  You should see your message content in clear-text as part of these fields.

**Comment by tsmethurst** [comment]
Hmm, not sure how much of a problem this is really. If someone has access to the application logs that means they're already on your server and likely have access to the GtS process and the database anyway. You could make a similar argument for TRACE level logging exposing database queries (which also contain stuff like this). I'm certainly up for changing the INFO level message you pasted there into a DEBUG level message, so that running on INFO level is a bit more private, but redacting all messages based on their audience might be a bit of a pain in the arse.

**Comment by NyaaaWhatsUpDoc** [comment]
at debug that is to be expected, but certainly not at info level. debug is very much a development / troubleshooting log level.

**Comment by nachtlied1** [comment]
Thanks, that makes sense. I wasn't sure if this was intentional at the `DEBUG` level, but at the `INFO` level I would probably expect less detail.

**Comment by decentral1se** [comment]
> ... I'm certainly up for changing the INFO level message you pasted there into a DEBUG level message, so that running on INFO level is a bit more private...
> ...at debug that is to be expected, but certainly not at info level...
> ...but at the INFO level I would probably expect less detail....

Opened https://github.com/superseriousbusiness/gotosocial/pull/1829 :+1:

**Comment by tsmethurst** [comment]
Hang on... reading this again, we already *do* provide less info when log-level is set to INFO rather than DEBUG, because of these lines:

https://github.com/superseriousbusiness/gotosocial/blob/fc524f8cf16761f949303a96bca14e63c46ceda8/internal/processing/fromfederator.go#L53-L57

and

https://github.com/superseriousbusiness/gotosocial/blob/fc524f8cf16761f949303a96bca14e63c46ceda8/internal/processing/fromclientapi.go#L43-L49

In other words, while the message might be logged at level INFO, we only actually add those extra fields when configured log-level is debug or higher (so, debug or trace basically). So this was actually never an issue, I believe :P

**Comment by tsmethurst** [comment]
I'm gonna close this issue, if that's alright, since those `info` level calls are very useful and don't reveal anything untoward, they only go into detail if log-level is set to debug, which is useful!
